onnxruntime/core/providers/webnn/builders/model_builder.cc

```diff
@@ -104,13 +104,15 @@ Status ModelBuilder::RegisterInitializers() {
     emscripten::val operand = emscripten::val::object();
     if (IsSupportedDataType(data_type, webnn_supported_data_types)) {
       ORT_RETURN_IF_NOT(SetWebnnDataType(desc, data_type), "Unsupported data type");
-      auto num_elements = SafeInt<size_t>(Product(tensor.dims()));
+      auto num_elements = SafeInt<size_t>(Product(shape));
       emscripten::val view = emscripten::val::undefined();
       std::byte* tensor_ptr = nullptr;
       if (tensor.has_raw_data()) {
         tensor_ptr = reinterpret_cast<std::byte*>(const_cast<char*>(tensor.raw_data().c_str()));
       } else {
-        std::vector<uint8_t> unpacked_tensor;
+        // Store temporary unpacked_tensor.
+        unpacked_tensors_.push_back({});
+        std::vector<uint8_t>& unpacked_tensor = unpacked_tensors_.back();
```

Comment:
Is this change related, or is it some other fix?

Reply:
Scalar needs to be unpacked from tensor, this fixes a bug that previous `unpacked_tensor` is a temporary argument, which will be destructed after this `else` branch.

So we need an global object to store it.

Reply:
It's an existing bug. :p